tools/utils/files.py

```diff
@@ -0,0 +1,16 @@
+import os
+from contextlib import contextmanager
+import logging
+
+@contextmanager
+def setdir(path):
```

Comment:
is there a better benefit of doing this? is this a python conversion?

Reply:
This is pretty common. See
* https://github.com/jaraco/path
* https://dev.to/teckert/changing-directory-with-a-python-context-manager-2bj8

Reply:
I could `chdir` in at the beginning and `chdir` out at the end, but I don't feel it pythonic enough.